plugins/dell-dock/fu-dell-dock-common.c

```diff
@@ -43,3 +43,20 @@ fu_dell_dock_set_power(FuDevice *device, guint8 target, gboolean enabled, GError
 
 	return fu_dell_dock_ec_modify_lock(parent, target, enabled, error);
 }
+
+const gchar *
+fu_dell_dock_get_instance_id(guint8 *type, DellDockComponent *dev_list, guint16 *vid, guint16 *pid)
+{
+	/* The last instance_id must be NULL */
+	for (guint i = 0; dev_list[i].instance_id != NULL; i++) {
+		if (*type != dev_list[i].dock_type)
+			continue;
+		if (dev_list[i].vid != 0 && vid != NULL && dev_list[i].vid != *vid)
+			continue;
+		if (dev_list[i].pid != 0 && pid != NULL && dev_list[i].pid != *pid)
+			continue;
```

Comment:
I think `dev_list[i].vid != 0` and `dev_list[i].pid != 0 ` are now unnecessary since the loop will break at the sentinel

Reply:
In a forthcoming PR will utilize the same instance query for mst & package devices, these devices are hardcoded with dock type without PID from the USB interface therefore the `continue` are here to guard the logic.
E.g.:  https://github.com/fwupd/fwupd/pull/7355/files#diff-cf3e2d3d7e960a425b0276b4438c42ef41d1d841fddafa0d256331654f303844R33

Reply:
Got it, then let's leave as is.